filebeat/inputsource/udp/server.go

```diff
@@ -0,0 +1,79 @@
+package udp
+
+import (
+	"net"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common/atomic"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Config options for the UDPServer
+type Config struct {
+	Host           string        `config:"host"`
+	MaxMessageSize int           `config:"max_message_size"`
+	Timeout        time.Duration `config:"timeout"`
+}
+
+// Server creates a simple UDP Server and listen to a specific host:port and will send any
+// event received to the callback method.
+type Server struct {
+	running  atomic.Bool
+	config   *Config
+	callback func(data []byte)
```

Comment:
I think the callback should include the source address (ip/port). This will allow for source metadata to be added to the event. Or in the future it could allow for per source buffering.